rixmp/source/R/class_ixmp.Platform.R

```diff
@@ -0,0 +1,108 @@
+ixmp.Platform <- methods::setRefClass("ixmp.Platform",
+    fields = c(.jobj = "jobjRef"),
+
+    #launch the Platform
+    methods = list(
+      initialize = function(dbprops=NULL, dbtype=NULL) {
+        "launch the Platform"
+
+        if (is.null(dbtype)) {
+          if (is.null(dbprops)) dbprops = 'default.properties'
+          if (!file.exists(dbprops)) {
+            dbprops = paste(ixmp_path, "/config/",dbprops, sep = '')
+            if (!file.exists(dbprops)) {
+              stop('no properties file ', dbprops , '!')
+            }
+          }
+          print(paste0("launching ixmp.Platform using config file at ",dbprops))
+          .jobj <<- rJava::new(rJava::J("at.ac.iiasa.ixmp.Platform"), "R", dbprops)
+        } else {
+          if (is.null(dbprops)) dbprops = paste(local_path, "/localdb/default", sep = '')
+            print(paste0("launching ixmp.Platform with local ",dbtype," database at "
+                      , dbprops))
+          .jobj <<- rJava::new(rJava::J("at.ac.iiasa.ixmp.Platform"), "R", dbprops, dbtype)
+        }
+
+        return(.self)
+      },
+
+      #initialize a new Scenario (structured model input data and solution)
+      # or get an existing Scenario from the IXMP database
+      Scenario = function(model, scen, version=NULL,
+                              scheme=NULL, annotation=NULL) {
+        "initialize a new Scenario (structured model input data and solution)
+        or get an existing Scenario from the IXMP database"
+        if (!is.null(version) && version=='new') {
+          jScen = .jobj$newScenario(model, scen, scheme, annotation)
+        } else if (!is.null(version)) {
+          jScen = .jobj$getScenario(model, scen, as.integer(version))
+        } else {
+          jScen = .jobj$getScenario(model, scen)
+        }
+
+        return(ixmp.Scenario(.jobj, model, scen, jScen))
+      },
+
+      units = function() {
+        "units function"
+        return(.jobj$getUnitList())
+      },
+      open_db = function(){
+        "(re-)open the database connection of the platform instance,
+        e.g., to continue working after using 'close_db()"
+        .jobj$openDB()
+      },
+
+      close_db = function(){
+        "close the database connection of the platform instance
+        this is important when working with local database files ('HSQLDB') "
+        .jobj$closeDB()
+      },
+
+      scenario_list = function(default=TRUE, model=NULL, scenario=NULL){
+        "Get a list of all dataobjects (timeseries)
+        and scenarios initialized in the IX Modeling Platform"
+        #for more complete doc
+        # Parameters
+        # ----------
+        # default : boolean, default True
+        # include only default model/scenario version (true) or all versions
+        # model : string
+        # the model name (optional)
+        # scen : string
+        # the scenario name (optional)
+
+        mod_scen_list = .jobj$getModelScenarioList(default, model, scenario)
```

Comment:
this function was renamed to `getScenarioList()` on the Java API, this causes an error in the R transport tutorial 2

Reply:
That's a big problem... I am trying to see what's wrong